### PR TITLE
Add several features to LaserZap

### PR DESCRIPTION
**Pull request by reaperrr**
- secondary beam for glow effect
- disable (visual) target-tracking
- support for inaccuracy
- support for blockability

I have some more, but I'll leave that for a follow-up.

### CODE REVIEW

File: OpenRA.Mods.Common/Projectiles/LaserZap.cs [hunk 1]
```diff
@@ -32,13 +33,42 @@ public class LaserZapInfo : IProjectileInfo
...
+		public readonly WDist Inaccuracy = WDist.Zero;
+
+		[Desc("Extra search radius beyond beam width. Required to ensure affecting actors with large health radius.")]
+		public readonly WDist TargetExtraSearchRadius = new WDist(1536);
+
+		[Desc("Beam can be blocked.")]
+		public readonly bool Blockable = false;
+
+		[Desc("Draw a second beam (for 'glow' effect).")]
+		public readonly bool SecondaryBeam = false;
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Projectiles/LaserZap.cs]: This will need a `TertiaryBeam` etc. so why not add another LaserBeam that doesn't do any damage?

Comment (GraionDilach) [inline OpenRA.Mods.Common/Projectiles/LaserZap.cs]: In his defense, TS/RA2 did have two lasers drawn at once - LaserInnerColor and LaserOuterColor defined that. (TS/RA2 also had LaserOuterSpread though.)

**Comment by abcdefg30** [comment]
Looks good to me. :+1: / :white_check_mark:

File: OpenRA.Mods.Common/Projectiles/LaserZap.cs (hunk 1 above)
Comment (Mailaender) [inline OpenRA.Mods.Common/Projectiles/LaserZap.cs]: I see.

File: OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs [hunk 2]
```diff
@@ -422,6 +422,13 @@ internal static void UpgradeWeaponRules(ModData modData, int engineVersion, ref
 						node.Key = "Speed";
 				}
 
+				// Rename LaserZap BeamDuration to just Duration
+				if (engineVersion < 20160919)
```
Comment (obrakmann) [inline OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs]: Please update the date.

**Review by obrakmann — CHANGES_REQUESTED** [review]
Seems fine to me, 👍 

Please remove the test case.

**Comment by reaperrr** [comment]
Removed test case and updated upgrade rule date.

**Review by obrakmann — APPROVED** [review]

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/56ace71e0b1e3b058a6345daa4e0a2aba718a62d)

**Pull request merged**